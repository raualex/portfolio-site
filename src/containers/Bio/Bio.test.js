import React from 'react';
import { Bio, mapStateToProps, mapDispatchToProps } from './';
import { shallow } from 'enzyme';

describe('Bio', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Bio selectedNav={'Bio'} />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });
});

describe('mapStateToProps function', () => {
  it('should return an object with selected Navigation item', () => {
    const mockState = {
      selectedNav: 'Bio',
      somethingElse: 'booyah'
    }
    const expected = {
      selectedNav: 'Bio'
    }
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
  });
});

describe('mapDispatchToProps function', () => {
  const mockDispatch = jest.fn()
  let mappedProps;

  beforeEach(() => {
    mappedProps = mapDispatchToProps(mockDispatch)
  });

  it('should call dispatch when setSelectedProj is called', () => {
    mappedProps.setSelectedProj()
    expect(mockDispatch).toHaveBeenCalled()
  });

  it('should call dispatch when setSelectedNav is called', () => {
    mappedProps.setSelectedNav()
    expect(mockDispatch).toHaveBeenCalled()
  });
});