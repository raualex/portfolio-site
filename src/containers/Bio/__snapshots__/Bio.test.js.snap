// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bio should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Bio
    selectedNav="Bio"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Software Engineer with an aptitude for solving complex problems. I love to construct complex, fully tested, UI systems based on a full understanding of the User Experience. I thrive in a dynamic, collaborative environment that utilizes an agile workflow to tackle challenges and accomplish collective goals. Most importantly, I wish to use my skills to serve a mission-driven company that helps people and communities in need.
        </p>,
        <p>
          Front End languages and libraries I work with: React, Redux, JavaScript/jQuery, CSS3, HTML5. I can execute asynchronous JavaScript to interact with RESTful APIs, and a PostgreSQL Back End. I can also build my own Back End using Node, Knex, Express and PostgreSQL. In addition, I have experience using: Microsoft Office/Excel and JIRA, as well as excellent organizational and communication skills, and a strong desire to make positive change in the world.
        </p>,
      ],
      "className": "bio-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Software Engineer with an aptitude for solving complex problems. I love to construct complex, fully tested, UI systems based on a full understanding of the User Experience. I thrive in a dynamic, collaborative environment that utilizes an agile workflow to tackle challenges and accomplish collective goals. Most importantly, I wish to use my skills to serve a mission-driven company that helps people and communities in need.",
        },
        "ref": null,
        "rendered": "Software Engineer with an aptitude for solving complex problems. I love to construct complex, fully tested, UI systems based on a full understanding of the User Experience. I thrive in a dynamic, collaborative environment that utilizes an agile workflow to tackle challenges and accomplish collective goals. Most importantly, I wish to use my skills to serve a mission-driven company that helps people and communities in need.",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Front End languages and libraries I work with: React, Redux, JavaScript/jQuery, CSS3, HTML5. I can execute asynchronous JavaScript to interact with RESTful APIs, and a PostgreSQL Back End. I can also build my own Back End using Node, Knex, Express and PostgreSQL. In addition, I have experience using: Microsoft Office/Excel and JIRA, as well as excellent organizational and communication skills, and a strong desire to make positive change in the world.",
        },
        "ref": null,
        "rendered": "Front End languages and libraries I work with: React, Redux, JavaScript/jQuery, CSS3, HTML5. I can execute asynchronous JavaScript to interact with RESTful APIs, and a PostgreSQL Back End. I can also build my own Back End using Node, Knex, Express and PostgreSQL. In addition, I have experience using: Microsoft Office/Excel and JIRA, as well as excellent organizational and communication skills, and a strong desire to make positive change in the world.",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Software Engineer with an aptitude for solving complex problems. I love to construct complex, fully tested, UI systems based on a full understanding of the User Experience. I thrive in a dynamic, collaborative environment that utilizes an agile workflow to tackle challenges and accomplish collective goals. Most importantly, I wish to use my skills to serve a mission-driven company that helps people and communities in need.
          </p>,
          <p>
            Front End languages and libraries I work with: React, Redux, JavaScript/jQuery, CSS3, HTML5. I can execute asynchronous JavaScript to interact with RESTful APIs, and a PostgreSQL Back End. I can also build my own Back End using Node, Knex, Express and PostgreSQL. In addition, I have experience using: Microsoft Office/Excel and JIRA, as well as excellent organizational and communication skills, and a strong desire to make positive change in the world.
          </p>,
        ],
        "className": "bio-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Software Engineer with an aptitude for solving complex problems. I love to construct complex, fully tested, UI systems based on a full understanding of the User Experience. I thrive in a dynamic, collaborative environment that utilizes an agile workflow to tackle challenges and accomplish collective goals. Most importantly, I wish to use my skills to serve a mission-driven company that helps people and communities in need.",
          },
          "ref": null,
          "rendered": "Software Engineer with an aptitude for solving complex problems. I love to construct complex, fully tested, UI systems based on a full understanding of the User Experience. I thrive in a dynamic, collaborative environment that utilizes an agile workflow to tackle challenges and accomplish collective goals. Most importantly, I wish to use my skills to serve a mission-driven company that helps people and communities in need.",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Front End languages and libraries I work with: React, Redux, JavaScript/jQuery, CSS3, HTML5. I can execute asynchronous JavaScript to interact with RESTful APIs, and a PostgreSQL Back End. I can also build my own Back End using Node, Knex, Express and PostgreSQL. In addition, I have experience using: Microsoft Office/Excel and JIRA, as well as excellent organizational and communication skills, and a strong desire to make positive change in the world.",
          },
          "ref": null,
          "rendered": "Front End languages and libraries I work with: React, Redux, JavaScript/jQuery, CSS3, HTML5. I can execute asynchronous JavaScript to interact with RESTful APIs, and a PostgreSQL Back End. I can also build my own Back End using Node, Knex, Express and PostgreSQL. In addition, I have experience using: Microsoft Office/Excel and JIRA, as well as excellent organizational and communication skills, and a strong desire to make positive change in the world.",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
